src/golang/cmd/server/handler/register_airflow_workflow.go

```diff
@@ -179,22 +256,26 @@ func (h *RegisterAirflowWorkflowHandler) Perform(ctx context.Context, interfaceA
 		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to create workflow.")
 	}
 
-	watchWorkflowArgs := &watchWorkflowArgs{
-		AqContext:  args.AqContext,
-		workflowId: workflowId,
-	}
+	if !args.isUpdate {
+		// Add watcher since this is a new workflow
+		watchWorkflowArgs := &watchWorkflowArgs{
+			AqContext:  args.AqContext,
+			workflowId: workflowId,
+		}
 
-	_, _, err = (&WatchWorkflowHandler{
-		Database:              h.Database,
-		WorkflowReader:        h.WorkflowReader,
-		WorkflowWatcherWriter: h.WorkflowWatcherWriter,
-	}).Perform(ctx, watchWorkflowArgs)
-	if err != nil {
-		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to add user who created the workflow to watch.")
+		_, _, err = (&WatchWorkflowHandler{
+			Database:              h.Database,
+			WorkflowReader:        h.WorkflowReader,
+			WorkflowWatcherWriter: h.WorkflowWatcherWriter,
+		}).Perform(ctx, watchWorkflowArgs)
+		if err != nil {
+			return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to add user who created the workflow to watch.")
+		}
 	}
 
 	return &registerAirflowWorkflowResponse{
-		Id:   workflowId,
-		File: string(airflowFile),
+		Id:       workflowId,
+		File:     string(airflowFile),
+		IsUpdate: args.isUpdate,
```

Comment:
why the new field is required in response?

Reply:
For the SDK, if its an update then we can tell the user that if they don't update the Airflow DAG file, new runs won't sync over properly.